modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/FieldsToDDMFormValuesConverterImpl.java

```diff
@@ -211,19 +211,18 @@ protected String getDDMFieldValueString(
 
 			fieldValue = valueDate.getTime();
 		}
-		else if (fieldValue instanceof Number) {
+		else if ((fieldValue instanceof Number) &&
+				 !(fieldValue instanceof Integer)) {
```

Comment:
@MarcellGyopos See this line. If you remove **!(fieldValue instanceof Integer)** from the condition then in line 225 an integer number will be formatted as a double including the decimal separator and a zero.